icu4c/source/i18n/double-conversion.h

```diff
@@ -503,16 +525,26 @@ class StringToDoubleConverter {
   //    StringToDouble("01239E45") -> 1239e45.
   //    StringToDouble("-infinity") -> NaN  // junk_string_value.
   //    StringToDouble("NaN") -> NaN  // junk_string_value.
+  //
+  //  flags = NO_FLAGS,
+  //  separator = ' ':
+  //    StringToDouble("1 2 3 4") -> 1234.0
+  //    StringToDouble("1  2") -> NaN // junk_string_value
+  //    StringToDouble("1 000 000.0") -> 1000000.0
+  //    StringToDouble("1.000 000") -> 1.0
+  //    StringToDouble("1.0e1 000") -> NaN // junk_string_value
   StringToDoubleConverter(int flags,
                           double empty_string_value,
                           double junk_string_value,
                           const char* infinity_symbol,
-                          const char* nan_symbol)
+                          const char* nan_symbol,
+                          uc16 separator = kNoSeparator)
       : flags_(flags),
         empty_string_value_(empty_string_value),
         junk_string_value_(junk_string_value),
         infinity_symbol_(infinity_symbol),
-        nan_symbol_(nan_symbol) {
+        nan_symbol_(nan_symbol),
+        separator_(separator) {
```

Comment:
The MSVC builds generate/output several warnings as `separator_` is a `uint16_t` but is seems to be consistently down-cast to a single `char` in multiple (all?) places. [This almost seems like a bug?].
I wonder if we should just make it a `char` in the header file as an ICU patch for now...?

Reply:
Good catch!  Looks like this is a bug in double-conversion.  I filed an upstream ticket:

https://github.com/google/double-conversion/issues/89

Changing the type to char in the header file seems like the least disruptive patch.  Done.

Reply:
Thanks for filing an upstream bug/ticket!